broker/src/main/java/io/zeebe/broker/system/partitions/ZeebePartitionHealth.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.broker.system.partitions;
+
+import io.zeebe.util.health.FailureListener;
+import io.zeebe.util.health.HealthMonitorable;
+import io.zeebe.util.health.HealthStatus;
+
+/**
+ * Reflects the health of ZeebePartition. The health is updated by ZeebePartition when role
+ * transitions either succeeded or failed.
+ */
+class ZeebePartitionHealth implements HealthMonitorable {
+  private HealthStatus healthStatus = HealthStatus.UNHEALTHY;
+  private final String name;
+  private FailureListener failureListener;
+
+  public ZeebePartitionHealth(final int partitionId) {
+    this.name = "ZeebePartition-" + partitionId;
+  }
+
+  @Override
+  public HealthStatus getHealthStatus() {
+    return healthStatus;
+  }
+
+  @Override
+  public void addFailureListener(final FailureListener failureListener) {
+    this.failureListener = failureListener;
+  }
+
+  void setHealthStatus(final HealthStatus healthStatus) {
+    final var previousStatus = this.healthStatus;
+    this.healthStatus = healthStatus;
+
+    if (previousStatus != healthStatus && failureListener != null) {
```

Comment:
Also, perhaps we can find a way to remove the duplication between this method, `CriticalComponentsHealthMonitor#calculateHealth` and `ZeebePartition#notifyFailureListenerAndUpdateMetrics`?

Reply:
> It's possible that the health status is updated, but that the failure listener is not made aware of this (because failure listener is still null), should we add a delayed onRecover/onFailure to the moment the failure listener is added?

I would like to keep it simple and let it recover in the next probe. Also in this case, we know the failureListener is not null. But the check is added only for safety.

Reply:
> Also, perhaps we can find a way to remove the duplication between this method, `CriticalComponentsHealthMonitor#calculateHealth` and `ZeebePartition#notifyFailureListenerAndUpdateMetrics`?

I removed 'notifyFailureListenerAndUpdateMetrics' and handled these cases directly in  `ZeebePartition#onRecovered` and `onFailure`.